src/dispatch.c

```diff
@@ -358,14 +358,14 @@ static void qd_dispatch_set_router_area(qd_dispatch_t *qd, char *_area) {
 void qd_dispatch_free(qd_dispatch_t *qd)
 {
     if (!qd) return;
+    qd_router_free(qd->router);
```

Comment:
You've made a significant change to the shutdown order here.  I think it would be less risky to just move the qd_dispatch_set_router_* from the beginning to the end.  Those are just freeing allocated memory slots.

Reply:
ok will do